src/components/Table/useSortState.ts

```diff
@@ -56,3 +56,26 @@ export function useSortState<R extends Kinded, S>(
 
   return [sortState, setSortKey];
 }
+
+// Exported for testing purposes
+export function deriveSortState<S>(
+  currentSortState: SortState<S> | undefined,
+  clickedKey: S,
+  initialSortState: SortState<S> | undefined,
+): SortState<S> | undefined {
+  const [currentKey, currentDirection] = currentSortState || [];
+
+  // If the current sort state is not defined, or clicking a new column, then sort ASC on the clicked key
+  if (!currentSortState || clickedKey !== currentKey) {
+    return [clickedKey, ASC];
+  }
+
+  // Otherwise when clicking the current column, toggle through sort states
+  if (currentDirection === ASC) {
+    // if ASC -> go to desc
+    return [clickedKey, DESC];
+  }
+
+  // Else, direction is already DESC, so revert to original sort value or undefined.
+  return initialSortState ? [...initialSortState] : undefined;
```

Comment:
Do we need the copy here? Could this even be just `return initialSortState`?

Reply:
💯 - I was slimming down from my initial logic and guess I stopped too early! Thanks for the thorough feedback as always.